sdk/cosmosdb/cosmos/test/internal/session.spec.ts

```diff
@@ -70,6 +71,96 @@ describe("New session token", function () {
   });
 });
 
+describe.only("Integrated Cache Staleness", async function (this: Suite) {
+  beforeEach(async function () {
+    await removeAllDatabases();
+  });
+  const dbId = addEntropy("maxIntegratedCacheTestDB");
+  const containerId = addEntropy("maxIntegratedCacheTestContainer");
+  const dedicatedGatewayMaxAge = 600000;
+  const client = new CosmosClient({
+    endpoint,
+    key: masterKey,
+    consistencyLevel: ConsistencyLevel.Eventual,
+    plugins: [
+      {
+        on: "request",
+        plugin: async (context, next) => {
+          it("Should check if the max integrated cache staleness header is set and the value is correct.", async function () {
+            if (context.headers["x-ms-consistency-level"]) {
+              if (
+                context.resourceType == ResourceType.item ||
+                context.resourceType == ResourceType.sproc
+              ) {
+                if (context.headers["x-ms-dedicatedgateway-max-age"]) {
+                  assert.strictEqual(
+                    context.headers["x-ms-dedicatedgateway-max-age"].valueOf(),
+                    dedicatedGatewayMaxAge
+                  );
+                } else {
+                  assert(
+                    context.headers["x-ms-dedicatedgateway-max-age"],
+                    "x-ms-dedicatedgateway-max-age is not set."
+                  );
+                  assert.ifError(context.headers["x-ms-dedicatedgateway-max-age"]);
+                }
+              } else {
+                assert(
+                  context.headers["x-ms-dedicatedgateway-max-age"],
+                  "Attempt to use x-ms-dedicatedgateway-max-age on a non-item request."
+                );
+                assert.ifError(context.headers["x-ms-dedicatedgateway-max-age"]);
+              }
+            } else {
+              assert(
+                context.headers["x-ms-consistency-level"],
+                "x-ms-consistency-level is not set."
+              );
+              assert.ifError(context.headers["x-ms-consistency-level"]);
+            }
+          });
+          const response = await next(context);
+          return response;
+        },
+      },
+    ],
+  });
+  const itemRequestFeedOptions = {
+    maxIntegratedCacheStalenessInMs: dedicatedGatewayMaxAge,
+  };
+  const { database } = await client.databases.createIfNotExists({
+    id: dbId,
+  });
+  const { container } = await database.containers.createIfNotExists({
+    id: containerId,
+  });
+
+  // Should pass with maxIntegratedCacheStalenessInMs and consistency level set.
+  await container.items.create({ id: "1" });
+  await container.item("1").read(itemRequestFeedOptions);
```

Comment:
Would it be possible to split this test into different it() blocks? Even if you have to define a new client with a different plugin, I think it will help test readability and debugging.

```ts
it("readItem should pass with maxIntegratedCacheStalenessInMs and consistency level set", async () => {
 const client = new CosmosClient({
    endpoint,
    key: masterKey,
    consistencyLevel: ConsistencyLevel.Eventual,
    plugins: [
      {
        on: "request",
        plugin: async (context, next) => {/** Only check for the specific conditions to pass with maxIntegratedCacheStalenessInMs and consistency level set*/}
      }]
   });
  await container.items.create({ id: "1" });
  await container.item("1").read(itemRequestFeedOptions);
});

it("redDocument should pass with maxIntegratedCacheStalenessInMs and consistency level set. ", async () => {
const client = new CosmosClient({
    endpoint,
    key: masterKey,
    consistencyLevel: ConsistencyLevel.Eventual,
    plugins: [
      {
        on: "request",
        plugin: async (context, next) => {/** Only check for the specific conditions to pass with maxIntegratedCacheStalenessInMs and consistency level set*/}
      }]
   });

  await container.items.readAll(itemRequestFeedOptions).fetchAll();
})

// etc...
```

Reply:
@jay-most  I'd like to hear what you think about the comment about splitting into individual `it`s. The usual pattern for tests is

```ts
describe("Test", () => {
   it("should do something", () => {});
   it("should do something else", () => {});
})
```

However, the current test is executing the `it` within the constructor plugins which may hurt readability and debugging. I'd suggest inverting it as suggested in my comment above